### ISSUE TITLE
Oil has a number of accessibility issues

### ISSUE THREAD

**Issue by seanrowlands** [as-ideas/oil#287]
Oil at present isn't accessible to a number of users:
* Keyboard users can't use the advanced modal to select specific purpose types, nor accept all/ decline all
* `span` used instead of `button` for the accept all/ decline all buttons.
* No clear content structure (lack of headings/sections)
* Third party vendor lists, should be ul with li not many divs
* Third party vendor expandable item, should use aria-expanded 
* Purpose toggle buttons not focusable and therefore not keyboard accessible
* Form controls should be appropriately labelled

https://www.gov.uk/guidance/accessibility-requirements-for-public-sector-websites-and-apps, has meant that public sector entities in the UK must ensure that their apps and websites are accessible inline with https://www.w3.org/TR/WCAG21/ AA standard.

We are working on a number of these issues in a fork of oil and hope to create a PR to resolve the majority of these issues, once completed.
